### PR TITLE
Generic clients

### PR DESCRIPTION
**Pull request by benblack86**
Added back information on generic clients. I'm trying to learn more about higher kinded types so this now makes sense! Whilst I was there, I changed a few `flatMap`s to` map`s since nothing needed to be flatted.

### CODE REVIEW

File: colossus-docs/src/main/scala/GenericClient.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  implicit val ioSystem = IOSystem()
+
+  def doTheThing[A[_]](client: HttpClient[A])(implicit async: Async[A]): A[String] = {
+    async.map(client.send(HttpRequest.get("/"))) { response =>
+      response.body.bytes.utf8String
+    }
+  }
+
+  // using future client
+  implicit val futureAsync = new FutureAsync()
```
Comment (DanSimon) [inline colossus-docs/src/main/scala/GenericClient.scala]: Hmm, you shouldn't have to be creating your own `FutureAsync`.  Well, at least that was my intention.  This stuff is currently used in [ClientFactory](https://github.com/tumblr/colossus/blob/master/colossus/src/main/scala/colossus/service/ServiceDSL.scala#L226), and notice that an `AsyncBuilder` is used.  

This is probably out of scope for this PR, but it should ideally be the case that all you need to do is import a sub-package or object containing some implicits, and let the builder do the work for you.

Comment (benblack86) [inline colossus-docs/src/main/scala/GenericClient.scala]: I saw the AsyncBuilder, but I couldn't see how to use it and there was no obvious implicits I should be importing.

**Comment by benblack86** [comment]
Added `codecov.yml`

**Comment by benblack86** [comment]
Also bumping SBT version since the previous version was causing problems with IntelliJ.

**Pull request merged**